src/autoscale.js

```diff
@@ -0,0 +1,191 @@
+import { bringToFront, getAppLink, waitForTime, times, round } from './utilities';
+
+const MAX_CONTAINERS = 10;
+const MIN_CONTAINERS = 2;
+
+const trySendAlertToSlack = ({ appLink, msgTitle, activeMetrics, activeMetricsByContainer }, options, slack) => {
+  const activeMetricsFormatted = {
+    pubSubResponseTime: `${round(activeMetrics.pubSubResponseTime)}ms`,
+    methodResponseTime: `${round(activeMetrics.methodResponseTime)}ms`,
+    memoryUsageByHost: `${round(activeMetrics.memoryUsageByHost)}MB`,
+    cpuUsageAverage: `${round(activeMetrics.cpuUsageAverage)}%`,
+    sessionsByHost: `${round(activeMetrics.sessionsByHost, 1)}%`,
+  };
+  const lastMetricsText = `${Object.entries(activeMetricsFormatted)
+    .map(([key, value]) => `*${key}*\n${value}`)
+    .join('\n')}`;
+  slack.alert({ text: `${appLink}\n${msgTitle}\n\n*Metrics*\n${lastMetricsText}\n` });
+};
+
+const checkAction = (action, rules, metrics, { andMode = true } = {}) => {
+  const when = rules[action];
+  const {
+    responseTimeAbove,
+    responseTimeBelow,
+    cpuAbove,
+    cpuBelow,
+    sessionsAbove,
+    sessionsBelow,
+  } = when || {};
+
+  const {
+    pubSubResponseTime,
+    methodResponseTime,
+    cpuUsageAverage,
+    sessionsByHost,
+  } = metrics;
+
+  let shouldRunAction = !!when;
+  if(!shouldRunAction) return false;
+
+  shouldRunAction = andMode;
+
+  let intermediateCheck = (pubSubResponseTime > responseTimeAbove && methodResponseTime > responseTimeAbove);
```

Comment:
What's with the non-listed formatting?

Reply:
Do you refer to apply prettier/eslint here? I'm not sure what you mean.